mypy/checker.py

```diff
@@ -5036,6 +5036,8 @@ def visit_if_stmt(self, s: IfStmt) -> None:
 
                 if_map, else_map = self.find_isinstance_check(e)
 
+                s.else_irrelevant_for_possibly_undefined = else_map is None
```

Comment:
So `else_irrelevant_for_possibly_undefined` is essentially equivalent to `else_unreachable` or `is_exhaustive`? I'm a bit confused by this naming